R/deconvolution_RL.R

```diff
@@ -43,23 +43,27 @@ deconvolution_RL <- function(
   # Check inputs
   stopifnot(is.vector(observed) & is.vector(times) & is.vector(p_delay))
   stopifnot(length(p_delay) <= length(observed))
+
+  # Pad the end of p_delay with 0s so
+  # it's the same length as the observations
   if(length(p_delay)<length(observed)){
-    # Pad the end of p_delay with 0s so it's the same length as the observations
     p_delay <- c(p_delay,
                  rep(0, length(observed)-length(p_delay)))
   }
   stopifnot(length(observed) == length(times) & length(observed) == length(p_delay))
+
   if(abs(sum(p_delay)-1) > 1e-6 & verbose){
     warning('p delay vector does not sum to 1. Extened max delay?')
   }
   observed <- na_to_0(observed)
 
   # From the delay multinomial, extract the mean and max delay
   # from infection to observation
-  mean_delay <- sum(p_delay * 0:(length(p_delay)-1))
-  max_delay <- utils::tail((0:length(p_delay))[p_delay>0], 1)
+  nd = length(p_delay)
+  mean_delay <- sum(p_delay * 0:(nd-1))
+  max_delay  <- utils::tail( (1:nd)[p_delay > 0], 1)
   orig_min_t <- min(times)
-  orig_ceiling <- observed[1]*10
```

Comment:
@davidchampredon what is the purpose of this hard-coded factor? just curious

Reply:
@papsti The RL algorithm needs a maximum value ("ceiling") for the values not observed at the very beginning of the time series (remember the deconvolution needs $f(t-k)$, but there is no information for $t\leq0$, so it needs to make an assumption). The original code from `EpiNow2` had a `*100` factor, which led to unrealistically large values for the inferred incidence. So I turn this factor down to `*10`. It's arbitrary, there may be a better way to deal with that...

Reply:
sounds good, i think your change is solid